src/components/EventCard.tsx

```diff
@@ -42,90 +54,157 @@ const EventCard = (event: Event): ReactElement => {
       })
     : [];
 
-  const [isRegistered, setIsRegistered] = useState(true);
+  //nuid of the corrently loggedIn user
+  const { userID } = useContext(LoginContext);
+  console.log(attendanceChange);
+  const [isRegistered, setIsRegistered] = useState(
+    attendanceChange ? false : true
+  );
+  const [errorType, setErrorType] = useState(0);
+  const [isOpen, setIsOpen] = useState(false);
+  const [createdAttendanceChange, setCreatedAttendanceChange] = useState({});
```

Comment:
You already commented this needs a type somewhere but that is something we should prioritize b4  merging